pkg/cmd/tracerunner.go

```diff
@@ -80,14 +106,21 @@ func (o *TraceRunnerOptions) Run() error {
 		if len(*pid) == 0 {
 			return fmt.Errorf("invalid pid found")
 		}
-		f, err := ioutil.ReadFile(programPath)
-		if err != nil {
-			return err
-		}
-		programPath = path.Join(os.TempDir(), "program-container.bt")
-		r := strings.Replace(string(f), "$container_pid", *pid, -1)
-		if err := ioutil.WriteFile(programPath, []byte(r), 0755); err != nil {
-			return err
+
+		if !o.isBcc {
+			f, err := ioutil.ReadFile(programPath)
+			if err != nil {
+				return err
+			}
+			programPath = path.Join(os.TempDir(), "program-container.bt")
+			r := substituteVariables(string(f), *pid)
+			if err := ioutil.WriteFile(programPath, []byte(r), 0755); err != nil {
+				return err
+			}
+		} else {
+			for i := range o.bccArgs {
+				o.bccArgs[i] = substituteVariables(o.bccArgs[i], *pid)
```

Comment:
One caveat I've discovered about this is when submitting the trace command it has to be single quoted to prevent bash from trying to interpolate it on the user side, e.g.

```bash
$ kubectl trace ... --bcc memleak -- -p '$container_pid'
```